shell/platform/glfw/key_event_handler.cc

```diff
@@ -5,21 +5,68 @@
 #include "flutter/shell/platform/glfw/key_event_handler.h"
 
 #include <iostream>
+#include <map>
 
 #include "flutter/shell/platform/common/cpp/client_wrapper/include/flutter/json_message_codec.h"
 
 static constexpr char kChannelName[] = "flutter/keyevent";
 
 static constexpr char kKeyCodeKey[] = "keyCode";
 static constexpr char kKeyMapKey[] = "keymap";
+static constexpr char kScanCodeKey[] = "scanCode";
+static constexpr char kModifiersKey[] = "modifiers";
 static constexpr char kTypeKey[] = "type";
+static constexpr char kToolkitKey[] = "toolkit";
+static constexpr char kUnicodeScalarValuesProduced[] =
+    "unicodeScalarValuesProduced";
+
+static constexpr char kLinuxKeyMap[] = "linux";
+static constexpr char kGLFWKey[] = "glfw";
 
-static constexpr char kAndroidKeyMap[] = "android";
 static constexpr char kKeyUp[] = "keyup";
 static constexpr char kKeyDown[] = "keydown";
 
 namespace flutter {
 
+// Converts a utf8 const char* to a 32 bit int. The Flutter framework accepts
+// only one 32 bit int, therefore, only up to 4 bytes are accepted. If it's
+// larger than 4 bytes, only the first 4 will be used.
```

Comment:
What happens if a multi-byte character straddles position 4? I didn't see any obvious handling of that; if I missed it maybe it just needs clear commenting at that point.